exonum-java-binding-time-service/src/main/java/com/exonum/binding/time/TimeSchemaProxy.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.time;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.serialization.Serializer;
+import com.exonum.binding.common.serialization.StandardSerializers;
+import com.exonum.binding.proxy.Cleaner;
+import com.exonum.binding.proxy.NativeHandle;
+import com.exonum.binding.proxy.ProxyDestructor;
+import com.exonum.binding.storage.database.View;
+import com.exonum.binding.storage.indices.EntryIndexProxy;
+import com.exonum.binding.storage.indices.ProofMapIndexProxy;
+import sun.reflect.generics.reflectiveObjects.NotImplementedException;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public class TimeSchemaProxy implements TimeSchema {
+
+  private static final Serializer<PublicKey> PUBLIC_KEY_SERIALIZER = StandardSerializers.publicKey();
+  private static final Serializer<ZonedDateTime> ZONED_DATE_TIME_SERIALIZER = StandardSerializers.zonedDateTime();
+
+  private final View dbView;
+
+  private TimeSchemaProxy(View dbView) {
+    this.dbView = dbView;
+  }
+
+  /**
+   * Constructs a schema proxy for a given dbView.
+   */
+  static TimeSchemaProxy newInstance(View dbView) {
```

Comment:
Why does it need to be a proxy of a native object, if that native object does not provide any useful operations?

Reply:
Well, this code is using approach with constructing time service collections by their names. I like this approach because it's much simpler (and in fact we don't need any Rust code at all), but from future perspective I think we need to encapsulate collections inside the Rust proxy.

Reply:
We will need that at one point, but I don't agree with putting some code that _pretends_ to do something, but doesn't until some time in the future (that might not come). Therefore I'd either keep things as simple as they need to be now, or finish the thing to create collections through a TimeSchema _in Rust_. The latter will also require adding appropriate factories to Java collection proxies, because now you can only create them by name, not from a random handle.